### PR TITLE
Add Evolmino

### PR DESCRIPTION
**Pull request by subaru-saito1**
I've implemeted a new type puzzle "Evolmino" ("シンカミノ: Shinkamino", in Japanese)
It's the latest puzzle appeared in Puzzle Publication Nikoli vol.182 (published on March 10th).

Rules and sample problem are here (Puzzle Square JP)
https://puzsq.logicpuzzle.app/puzzle/106378

### CODE REVIEW

**Comment by x-sheep** [comment]
Thanks for the Pull Request! This genre looks very interesting.

I'll do a code review at a later date, but I have a few points, some of which are from our testers:

* [ ] I'd like a separate one-button and two-button mouse input. See the puzzles with `use: true` inside `MouseEvent` for an example.
* [ ] Clicking the mouse in Edit mode cycles between Invalid Cell, Rectangle and Empty. This makes it impossible to add given Rectangles on top of an arrow using the left mouse button. I'd like the order of items to be reversed.
* [ ] I'd like to be able to expand an existing arrow by dragging from the tail end.

**Comment by subaru-saito1** [comment]
Thanks for feedback!
I'll try to implement them later!

**Comment by subaru-saito1** [comment]
I've fixed cell and arrow input UI!

- Implement switching 2-button/1-button mode both play and edit mode.
- Swap left and right click cycle of cell input (when in play mode + 1-button mode)
- Improve arrow input UI to be able to expand reverse direction from arrow tail.

File: src/variety/evolmino.js [hunk 1]
```diff
@@ -0,0 +1,870 @@
...
+					} else {
+						g.fillRectCenter(px, py, this.bw + lm, lm);
+					}
+				} else {
+					g.vhide();
+				}
+				// arrow part
+				g.vid = "b_arrow_" + border.id;
+				if (!!color && dir !== 0) {
+					if (dir === 1) {
```
Comment (x-sheep) [inline src/variety/evolmino.js:436]: Can you use the `LT` `RT` `UP` `DN` constants here?

File: test/script/evolmino.js [hunk 2]
```diff
@@ -0,0 +1,154 @@
+ui.debug.addDebugData("evolmino", {
+	url: "5/5/i71027000222k44j23344",
+	failcheck: [
+		[
+			"arInvalid",
```
Comment (x-sheep) [inline test/script/evolmino.js:5]: I think this check doesn't need to be shown on the rules page. Can you add `skiprules: true`?

**Review by x-sheep — CHANGES_REQUESTED** [review]

**Comment by subaru-saito1** [comment]
Thank you for reveiwing! I fixed them.

**Review by x-sheep — APPROVED** [review]
This implementation looks good to me. I only have a few minor points, but I can also apply these changes myself if needed.

I'm planning to release this genre on Monday. Thank you for your work 🎉

**Comment by x-sheep** [comment]
@subaru-saito1 A different question unrelated to Evolmino:

Do you still have the PDF's of the unused Omopa genres that used to be on your blog? I think some of them would be interesting to add to puzz.link 👀

**Comment by subaru-saito1** [comment]
@x-sheep 
I reuploaded those Omopa genres here: https://note.com/subaru_saito/n/ned95450dd181
I'll be add some problems and translate into English at a later date.

**Comment by x-sheep** [comment]
[Evolmino is now available.](https://puzz.link/p?evolmino)

If you want your existing puzzles to be picked up by [the database](https://puzz.link/db/), you'll have to Tweet them again with the puzz.link URL.

**Pull request merged**